### PR TITLE
extracting metadata is a user error not a compiler issue

### PR DESCRIPTION
**Pull request by cc10512**

### CODE REVIEW

File: testdata/p4_16_errors/nasty_meta.p4 [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/* -*- P4_14 -*- */
```
Comment (mbudiu-vmw) [inline testdata/p4_16_errors/nasty_meta.p4]: this is in the wrong folder, it should be in p4_14_errors

File: testdata/p4_16_errors_outputs/nasty_meta.p4-stderr [hunk 2]
```diff
@@ -0,0 +1,4 @@
+nasty_meta.p4(6):syntax error, unexpected {, expecting (
```
Comment (mbudiu-vmw) [inline testdata/p4_16_errors_outputs/nasty_meta.p4-stderr]: and this one should be deleted

File: testdata/p4_14_errors_outputs/nasty_meta.p4-stderr [hunk 3]
```diff
@@ -0,0 +1,4 @@
+nasty_meta.p4(6):syntax error, unexpected {, expecting (
```
Comment (mbudiu-vmw) [inline testdata/p4_14_errors_outputs/nasty_meta.p4-stderr]: Something is wrong, this is not the error message that you should expect.

Comment (cc10512) [reply testdata/p4_14_errors_outputs/nasty_meta.p4-stderr]: thanks for catching this. updated to the proper error message.

File: testdata/p4_14_errors_outputs/nasty_meta.p4-stderr [hunk 4]
```diff
@@ -0,0 +1,4 @@
+nasty_meta.p4(6): [--Werror=invalid] error: struct ethernet_t: Invalid expected a header
```
Comment (mbudiu-vmw) [inline testdata/p4_14_errors_outputs/nasty_meta.p4-stderr]: Sorry to be such a nuisance, but I think that the error should point to the extract and not to the header type declaration.

Comment (cc10512) [reply testdata/p4_14_errors_outputs/nasty_meta.p4-stderr]: No nuisance, I actually appreciate it! Better error messages are very important.

**Comment by cc10512** [comment]
@mbudiu-vmv I hope the message is clear now.

**Pull request merged**